packages/service-library/tests/fastapi/test_long_running.py

```diff
@@ -0,0 +1,195 @@
+"""
+Showcases/tests an example of long running tasks.
+
+How these tests works:
+- setup a FastAPI server and lunch it in the background.
+- client directly makes requests to the FastAPI background client.
+
+"""
+
+# pylint: disable=redefined-outer-name
+
+import asyncio
+import subprocess
+import sys
+from pathlib import Path
+from typing import AsyncIterator, Callable, Final
+
+import pytest
+from asgi_lifespan import LifespanManager
+from fastapi import Depends, FastAPI, status
+from httpx import AsyncClient, HTTPError
+from pydantic import AnyHttpUrl, parse_obj_as
+from servicelib.fastapi.long_running import client, server
+
+CURRENT_FILE = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve()
+CURRENT_DIR = CURRENT_FILE.parent
+
+ITEM_PUBLISH_SLEEP: Final[float] = 0.1
+
+# NOTE: `mock_server_app` needs to be defined at module level
```

Comment:
What about?
```python

def create_mock_app():
   app = FastAPI(title="app containing the server")
   server.setup(app)

   @app.get("/")
   def health() -> None:
       ...
    # etc   
    return app

@pytest.fixture
async def run_server(get_unused_port: Callable[[], int]) -> AsyncIterator[AnyHttpUrl]:
    port = get_unused_port()
    with subprocess.Popen(
        [
            "uvicorn",
            f"{CURRENT_FILE.stem}:mock_server_app",
    # ...

if __name__ == "__main__":
   mock_server_app = create_mock_app()
```

Reply:
Will give this a try!

Reply:
So you can get rid of the `application` global  if you use the `--factory`.

The `if __name__ == "test_long_running":` did not look too nice to me.

Reply:
yes, i did a mistake. ``uvicorn`` imports the module. My bad. forget about this. sorry!